### PR TITLE
Add Heartbeat to Messaging Bridge

### PR DESCRIPTION
**Pull request by kentdr**
- Related to : https://github.com/Particular/NServiceBus.MessagingBridge/issues/210

### CODE REVIEW

File: src/NServiceBus.MessagingBridge/Configuration/HostApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -22,15 +23,16 @@ public static IHostApplicationBuilder UseNServiceBusBridge(this IHostApplication
...
-            builder.Services.AddSingleton(sp => bridgeConfiguration.FinalizeConfiguration(sp.GetRequiredService<ILogger<BridgeConfiguration>>()));
-            builder.Services.AddSingleton(deferredLoggerFactory);
-            builder.Services.AddSingleton<IHostedService, BridgeHostedService>();
-            builder.Services.AddSingleton<IStartableBridge, StartableBridge>();
-            builder.Services.AddSingleton<EndpointProxyFactory>();
-            builder.Services.AddSingleton<SubscriptionManager>();
-            builder.Services.AddSingleton<EndpointRegistry>();
-            builder.Services.AddSingleton<IEndpointRegistry>(sp => sp.GetRequiredService<EndpointRegistry>());
-            builder.Services.AddSingleton<IMessageShovel, MessageShovel>();
+            _ = builder.Services.AddSingleton(sp => bridgeConfiguration.FinalizeConfiguration(sp.GetRequiredService<ILogger<BridgeConfiguration>>()));
```
Comment (awright18) [inline src/NServiceBus.MessagingBridge/Configuration/HostApplicationBuilderExtensions.cs:26]: this seems like something and IDE did, and is not likely what we want.

File: src/NServiceBus.MessagingBridge/Configuration/HostBuilderExtensions.cs [hunk 2]
```diff
@@ -33,22 +34,23 @@ public static IHostBuilder UseNServiceBusBridge(this IHostBuilder hostBuilder, A
...
-
-                serviceCollection.AddSingleton(deferredLoggerFactory);
-                serviceCollection.AddSingleton<IHostedService, BridgeHostedService>();
-                serviceCollection.AddSingleton<IStartableBridge, StartableBridge>();
-                serviceCollection.AddSingleton<EndpointProxyFactory>();
-                serviceCollection.AddSingleton<SubscriptionManager>();
-                serviceCollection.AddSingleton<EndpointRegistry>();
-                serviceCollection.AddSingleton<IEndpointRegistry>(sp => sp.GetRequiredService<EndpointRegistry>());
-                serviceCollection.AddSingleton<IMessageShovel, MessageShovel>();
+                _ = serviceCollection.AddSingleton(sp => bridgeConfiguration.FinalizeConfiguration(sp.GetRequiredService<ILogger<BridgeConfiguration>>()));
```
Comment (awright18) [inline src/NServiceBus.MessagingBridge/Configuration/HostBuilderExtensions.cs:43]: this seems like something and IDE did, and is not likely what we want.

Comment (awright18) [reply src/NServiceBus.MessagingBridge/Configuration/HostBuilderExtensions.cs:43]: It's UGLY. the other options is removing all the repeated serviceCollection. and chain the AddSignleton calls

Comment (awright18) [reply src/NServiceBus.MessagingBridge/Configuration/HostBuilderExtensions.cs:43]: I apologize, that was rude, what I meant to say is I haven't seen discards used in this context in applications before, and it gives a somewhat jarring impression.

File: src/NServiceBus.MessagingBridge/Utils/DeterministicGuid.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+﻿#nullable enable
+
+namespace NServiceBus.MessagingBridge;
+
+using System;
+using System.Buffers;
+using System.Security.Cryptography;
+using System.Text;
+
+static class DeterministicGuid
```
Comment (dvdstelt) [inline src/NServiceBus.MessagingBridge/Utils/DeterministicGuid.cs:10]: If you got the code for this from somewhere, we usually link back to the source. Also pay attention to licenses that the original source code might have.

**Review by dvdstelt — APPROVED** [review]

File: src/NServiceBus.MessagingBridge/Utils/DeterministicGuid.cs (hunk 3 above)
Comment (awright18) [reply src/NServiceBus.MessagingBridge/Utils/DeterministicGuid.cs:10]: Thanks, @dvdstelt. This was in NServiceBus.Core and didn't seem to be lifted from elsewhere.

**Comment by awright18** [comment]
closed in favor of 
- #387

**Pull request closed without merging**